pina/geometry/union.py

```diff
@@ -88,11 +95,25 @@ def sample(self, n, mode='random', variables='all'):
                 1000
         """
         sampled_points = []
-        for geometry in self.geometries:
-            points = geometry.sample(
-                int(n/len(self.geometries)), mode, variables)
+        remainder = n % len(self.geometries)
+
+        for i, geometry in enumerate(self.geometries):
+            num_points = n // len(self.geometries)
```

Comment:
this can be moved out of the for loop, otherwise we calculate it at each iteration. Please also add some comments like  `# calculate remainder and number of point`, `# sampling points` so it's more maintainable in the future.